File: frontend/src/api/customerSuccess.js
import { api } from './client'

// ---- Módulo de Clientes (lista/resumo agregados) -------------------------------

export function listClientes({
  csFase, csResponsavelId, busca, healthScoreMax, renovacaoAteDias, somenteEmRisco,
} = {}) {
  return api
    .get('/clientes', {
      params: {
        cs_fase: csFase, cs_responsavel_id: csResponsavelId, busca,
        health_score_max: healthScoreMax, renovacao_ate_dias: renovacaoAteDias,
        somente_em_risco: somenteEmRisco || undefined,
      },
    })
    .then((res) => res.data)
}

export function getClientesResumo() {
  return api.get('/clientes/resumo').then((res) => res.data)
}

// ---- Sub-recursos por empresa ---------------------------------------------------

export function setCompanyCs(id, csResponsavelId) {
  return api.patch(`/companies/${id}/cs`, { cs_responsavel_id: csResponsavelId }).then((res) => res.data)
}

export function setCompanyCsFase(id, csFase) {
  return api.patch(`/companies/${id}/cs-fase`, { cs_fase: csFase }).then((res) => res.data)
}

export function getCompanyHealth(id) {
  return api.get(`/companies/${id}/health`).then((res) => res.data)
}

export function createCheckin(id, data) {
  return api.post(`/companies/${id}/checkins`, data).then((res) => res.data)
}

export function listOnboarding(id) {
  return api.get(`/companies/${id}/onboarding`).then((res) => res.data)
}

export function setOnboardingItemStatus(companyId, itemId, status) {
  return api.patch(`/companies/${companyId}/onboarding/${itemId}`, { status }).then((res) => res.data)
}

File: app/api/v1/customer_success.py
"""Rotas do módulo de Clientes (Customer Success) — listagem e resumo agregado.
Sub-recursos por empresa (cs-fase, health, onboarding, check-ins) ficam em
app/api/v1/companies.py. Ver docs/PLANO_CUSTOMER_SUCCESS.md."""
from uuid import UUID

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.core.deps import get_current_user
from app.models.user import User
from app.schemas.company import ClienteRead, ClientesResumo
from app.services.company import CompanyService

router = APIRouter(prefix="/clientes", tags=["Customer Success"])


@router.get("", response_model=list[ClienteRead])
def list_clientes(
    cs_fase: str | None = None,
    cs_responsavel_id: UUID | None = None,
    busca: str | None = None,
    health_score_max: int | None = None,
    renovacao_ate_dias: int | None = None,
    somente_em_risco: bool = False,
    _: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    return CompanyService(db).list_clientes(
        cs_fase=cs_fase, cs_responsavel_id=cs_responsavel_id, busca=busca,
        health_score_max=health_score_max, renovacao_ate_dias=renovacao_ate_dias,
        somente_em_risco=somente_em_risco,
    )


@router.get("/resumo", response_model=ClientesResumo)
def get_clientes_resumo(_: User = Depends(get_current_user), db: Session = Depends(get_db)):
    return CompanyService(db).resumo_clientes()
